### PR TITLE
Bugfix #168172498 – Improve performance for sitemaps

### PR DESCRIPTION
**Pull request by alfonsomunozpomer**
These changes include:
1. Do all Solr-related queries using stream expressions with our SolrCloud mini-framework.
2. Cache results.
3. Make controller and `SitemapWriter` stateless (no need for `@Scope("request")`).
4. Add an experiments `sitemap.xml`, so we don’t need to include `/experiments` and `/plant/experiments` on every species sitemap, which is kind of weird.

Goes with https://github.com/ebi-gene-expression-group/atlas-web-core/pull/17.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/experimentimport/GxaExperimentCrud.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import org.apache.commons.lang3.tuple.Pair;
 import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
```
Comment (lingyun1010) [inline src/main/java/uk/ac/ebi/atlas/experimentimport/GxaExperimentCrud.java]: Stale import.

File: src/main/java/uk/ac/ebi/atlas/experimentpage/differential/topgenes/DifferentialExperimentTopGenesDao.java [hunk 2]
```diff
@@ -32,8 +32,8 @@ public TupleStreamer createForDifferentialNonSpecific(String experimentAccession
 //                        experimentAccession, preferences);
```
Comment (lingyun1010) [inline src/main/java/uk/ac/ebi/atlas/experimentpage/differential/topgenes/DifferentialExperimentTopGenesDao.java]: Are those `solrQuery` still useful or we could delete them? This results in a stale field `experimentAccession`.

File: src/test/java/uk/ac/ebi/atlas/sitemaps/SitemapWriterIT.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    SpeciesPropertiesTrader speciesPropertiesTrader;
+
+    @Test
+    void sitemapWriterIsAUtilityClassAndCannotBeInstantiated() {
+        assertThatExceptionOfType(UnsupportedOperationException.class)
+                .isThrownBy(SitemapWriter::new);
+    }
+
+//    @Test
+//    void testWriteOneGene() throws Exception {
```
Comment (lingyun1010) [inline src/test/java/uk/ac/ebi/atlas/sitemaps/SitemapWriterIT.java]: Why this test is commented?

File: src/test/java/uk/ac/ebi/atlas/sitemaps/SitemapDaoIT.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    @Test
+    void publicBioentityIdentifiersIsCaseInsensitive() {
+        assertThat(subject.getBioentityIdentifiersInPublicExperiments("Homo sapiens"))
+                .isEqualTo(subject.getBioentityIdentifiersInPublicExperiments("Homo_sapiens"))
+                .isEqualTo(subject.getBioentityIdentifiersInPublicExperiments("hOmO SapIEns"))
+                .isNotEmpty();
+    }
+
+    @Test
+    void unknownSpeciesReturnsEmpty() {
```
[the remark this answers is not in the document]
Comment (alfonsomunozpomer) [reply src/test/java/uk/ac/ebi/atlas/sitemaps/SitemapDaoIT.java]: Changed to `unknownOrEmpty...`

File: src/main/java/uk/ac/ebi/atlas/experimentimport/GxaExperimentCrud.java (hunk 1 above)
Comment (alfonsomunozpomer) [reply src/main/java/uk/ac/ebi/atlas/experimentimport/GxaExperimentCrud.java]: Removed.

**Pull request merged**